dll/shellext/fontext/CFontExt.cpp

```diff
@@ -262,7 +262,53 @@ STDMETHODIMP CFontExt::CompareIDs(LPARAM lParam, PCUIDLIST_RELATIVE pidl1, PCUID
     if (!fontEntry1 || !fontEntry2)
         return E_INVALIDARG;
 
-    int result = (int)fontEntry1->Index - (int)fontEntry2->Index;
+    int result;
+    DWORD sortMode = lParam & 0xFFFF0000;
+    DWORD column = lParam & 0x0000FFFF;
+    if (sortMode == SHCIDS_ALLFIELDS)
+    {
+        UNIMPLEMENTED;
+        result = (int)fontEntry1->Index - (int)fontEntry2->Index;
+    }
+    else
+    {
+        auto info1 = g_FontCache->Find(fontEntry1);
+        auto info2 = g_FontCache->Find(fontEntry2);
+
+        if (!info1 || !info2)
+        {
+            ERR("Unable to find font %S or %S in cache!\n", fontEntry1->Name, fontEntry2->Name);
+            return E_INVALIDARG;
+        }
+
+        switch (column)
+        {
+        case 0xffff:
+            /* ROS bug? */
```

Comment:
Maybe could go instead with the `default` case?

Reply:
no, because this happens and then we would not sort at all in ros.